asm/f64/stubs_test.go

```diff
@@ -0,0 +1,221 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+package f64
+
+import (
+	"math"
+	"runtime"
+	"testing"
+)
+
+var (
+	nan = math.NaN()
+	inf = math.Inf(1)
+)
+
+func same(a, b float64) bool {
+	return !(a != b && !math.IsNaN(a) && !math.IsNaN(b) || (math.IsNaN(a) != math.IsNaN(b)))
+}
+
+func TestAdd(t *testing.T) {
+	for j, v := range []struct {
+		dst, src, expect []float64
+	}{
+		{[]float64{1}, []float64{0}, []float64{1}},
+		{[]float64{1, 2, 3}, []float64{1}, []float64{2, 2, 3}},
+		{[]float64{}, []float64{}, []float64{}},
+		{[]float64{1}, []float64{nan}, []float64{nan}},
+		{[]float64{8, 8, 8, 8, 8},
+			[]float64{2, 4, nan, 8, 9},
+			[]float64{10, 12, nan, 16, 17}},
+		{[]float64{0, 1, 2, 3, 4},
+			[]float64{-inf, 4, nan, 8, 9},
+			[]float64{-inf, 5, nan, 11, 13}},
+	} {
+		Add(v.dst, v.src)
```

Comment:
This points to a need to have sentinel values in the tests, since we expect that subslices will be inputs to many if not all routines here. Having sentinels necessarily involves this kind of input.